packages/block-library/src/group/deprecated.js

```diff
@@ -18,6 +18,12 @@ const deprecated = [
 			customBackgroundColor: {
 				type: 'string',
 			},
+			anchor: {
+				type: 'string',
+				source: 'attribute',
+				attribute: 'id',
+				selector: '*',
+			},
```

Comment:
This means the hooks are not executed on the deprecated version of the block. Should we fix the root issue instead. I assume we still have issues for classNames and alignments.

Reply:
Yep. Not sure how this came about. I was concerned I caused it in https://github.com/WordPress/gutenberg/pull/12757, but that doesn't seem to be the case.

It's `addFilter( 'blocks.registerBlockType', 'core/anchor/attribute', addAttribute );` that isn't being run for the deprecation.

I'll do some more digging.

Reply:
@youknowriad From what I can tell this doesn't seem the hooks have always behaved this way. I haven't been able to pinpoint where a regression might have been introduced.

There's an alternative solution on https://github.com/WordPress/gutenberg/pull/16348 with some more details.

Reply:
I'm pretty sure I accounted for these when I initially built the deprecation support. (At least, I thought about it, I don't remember how well I tested it)

Reply:
I think the alternative fix is the right now, but we need to test it properly (maybe with some third-party plugins)

Reply:
Another option would be to add a new and entirely separate filter, something like `blocks.registerBlockType.deprecated` that allows filtering of each deprecation. Then implementing that filter for each of the supports settings that need it.

That would remove the worry about plugins being affected.

Reply:
I'd appreciate some thoughts here @gziolo @aduth @mcsf

Reply:
> I'm pretty sure I accounted for these when I initially built the deprecation support. (At least, I thought about it, I don't remember how well I tested it)

Can we `checkout` and `run` a couple of old commits and get a definitive answer on whether we did have deprecation support? I think any decisions in this issue, especially as regards third parties and compatibility, highly depend on that. @talldan or @youknowriad, I don't know which of you would rather have a look.

Reply:
I don't recall whether this ever worked, but the approach in #16348 seems reasonable to me. It might become a bit tricky in cases where core or plugins filter `registerBlockType` with certain assumptions about "current" settings (i.e. current at the time they're implemented). I still think it's a safer default for future compatibility that these deprecations be treated as any block registration normally would be. Probably also wiser for said core / plugins filtering to not assume anything specific about current settings of a block (or at least not without conditions to guarantee those assumptions).

Reply:
@mcsf I've just tested 5.2, and it was still an issue back then, so there's a good chance it has never worked. 

Other supports properties don't seem to trigger a validation error—for example `align` 
options fall back to being a custom class name. The custom class name setting itself doesn't need an attribute definition. I think that explains why the issue wasn't noticed before.

Reply:
I've tested coblocks, stackable and atomic blocks with #16348 and they seem to work fine. Coblocks seems to have an unrelated issue with one of its blocks being registered twice.

I'll close this PR since the other one has been approved.